doc/api/index.md

```diff
@@ -29,6 +29,7 @@
 * [Domain](domain.md)
 * [Errors](errors.md)
 * [Events](events.md)
+* [Fetch](fetch.md)
```

Comment:
I am a bit concerned that folks will misinterpret this as being a full fetch implementation...

Reply:
What would be the best way to introduce this without implementing all of fetch?

I could make a new `feature/fetch` branch and we can merge all fetch work to that until its completely ready. I'm really eager to start integrating `undici-fetch` with the `streams/web` api